config.go

```diff
@@ -535,7 +535,17 @@ func loadConfig() (*config, []string, error) {
 	// All data is specific to a network, so namespacing the data directory
 	// means each individual piece of serialized data does not have to
 	// worry about changing names per network and such.
+	//
+	// Check for old versions of testnet directories here since the network
+	// specific DataDir will be used after this.
 	cfg.DataDir = cleanAndExpandPath(cfg.DataDir)
+	var oldTestNet bool
+	oldDataDir := filepath.Join(cfg.DataDir, "testnet")
+	if cfg.TestNet {
```

Comment:
I couldn't decide if it should only happen with --testnet or not so I just put it in that way and figured if someone felt strongly they would say something.  I really couldn't decide so i'll go whichever way you want there.